pkg/asset/machines/gcp/gcpmachines.go

```diff
@@ -0,0 +1,183 @@
+// Package gcp generates Machine objects for gcp.
+package gcp
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+// GenerateMachines returns manifests and runtime objects to provision control plane nodes using CAPI.
+func GenerateMachines(installConfig *installconfig.InstallConfig, infraID string, pool *types.MachinePool, imageName string) ([]*asset.RuntimeFile, error) {
+	var result []*asset.RuntimeFile
+	if poolPlatform := pool.Platform.Name(); poolPlatform != gcptypes.Name {
+		return nil, fmt.Errorf("non-GCP machine-pool: %q", poolPlatform)
+	}
+	mpool := pool.Platform.GCP
+
+	total := int64(1)
```

Comment:
Shouldn't the initial total value be 0?
```suggestion
	total := int64(0)
```

Reply:
`total` is the count of control-plane nodes if the machine pool replica count is not set. so 1 makes sense. 

this code is copy-pasted from the machine manifest generation and in practice I think the machine pool replica is always set because it has a default in `pkg/types`